File: Normal/0001/main.ts
function twoSum(nums: number[], target: number): number[] {
    let hashmap: Map<number, number> = new Map()
    let [value, i2] = [0, 0]
    for (let i1 in nums) {
        value = target - nums[i1]
        i2 = hashmap.get(value) ?? 0

        if (i2 != undefined) {
            return [i2, Number(i1)]
        }
        hashmap.set(nums[i1], Number(i1))
    }
    return []
};

File: Normal/0001/main.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> hashmap;
        for (int i = 0; i < nums.size(); i++)
        {
            auto value = hashmap.find(target - nums[i]);
            if (value != hashmap.end())
            {
                return {value->second, i};
            }
            hashmap[nums[i]] = i;
        }
        return {};
    }
};